### DISCUSSION TITLE
Utopia

### DISCUSSION THREAD

**Discussion by AExusiai** [bcholmes/StarTrek2d20#70] [Q&A]
Mr. Holmes, 
How long do you think it will take you to input the data from UP book? And while I have no clue about coding, is there anything I can do to help?

**Comment by bcholmes** [comment]
There are a couple of answers to the question of "how long". 

First, Modiphius doesn't want material from new sourcebooks in the app until the books are out of "pre-order" -- so at the very least, I wait until the PDFs are available as an independent purchase (my guess is that's probably somewhere in the November timeframe, but I don't know for sure). Obviously people who have the pre-order are hungry for stuff sooner than that, but Modiphius does own the material and I think they've been quite reasonable in allowing so much of their rules to be in the app for free.

Second, I tend to only buy PDFs, so I'm not likely to have a copy of Utopia Planitia until the PDF purchase is available. I've generally bought things the same day that they're announced. I'll get the PDF and it'll probably take a week or so after that. (The Player's Guide required *a lot* of rework, and that was a bit slower; I don't know how different the new rules are yet, but they're probably nowhere near the same amount of rework). 

As for what can you do? I'll give that some thought. 

Maybe you can help with some simple planning. I usually start with a simple list of the changes. How accurate is this list (I'm basing this on the table of contents that's been shown):

1. A bunch of new spaceframes
2. New talents (?)
3. New weapons (?)
4. Spaceframe creation rules
5. Optional ship rules (how many? Do they affect ship creation)
6. Space station rules

Is there other stuff?

**Reply by Felderburg** [reply]
Well, based on the forum post where it is noted that all spaceframes in UP were modified to be in-line with the new construction rules, it seems like that may cause issues. My suggestion would be to add a choice right after picking starship on the "What do you want to create?" screen there is a choice of 'pre-UP' or 'post-UP' creation. Or maybe that's determined by selecting (or not) the book even before that step? Either way, making it very clear that there are (apparently) 2 different versions of pre-made ships.

(7. Pre- vs. post-UP spaceframes)


Also, as seen in #62 & #60 , there could very well be more one-off type of spaceframes, so figuring out how to incorporate them might be worth looking at in the rework. (Or any other 'non-traditional' or outside-the-rules type of ship. Maybe an NPC ship section? Although that might stray too far into copying the books.)

**Reply by AExusiai** [reply]
> There are a couple of answers to the question of "how long".
> 
> First, Modiphius doesn't want material from new sourcebooks in the app until the books are out of "pre-order" -- so at the very least, I wait until the PDFs are available as an independent purchase (my guess is that's probably somewhere in the November timeframe, but I don't know for sure). Obviously people who have the pre-order are hungry for stuff sooner than that, but Modiphius does own the material and I think they've been quite reasonable in allowing so much of their rules to be in the app for free.
> 
> Second, I tend to only buy PDFs, so I'm not likely to have a copy of Utopia Planitia until the PDF purchase is available. I've generally bought things the same day that they're announced. I'll get the PDF and it'll probably take a week or so after that. (The Player's Guide required _a lot_ of rework, and that was a bit slower; I don't know how different the new rules are yet, but they're probably nowhere near the same amount of rework).
> 
> As for what can you do? I'll give that some thought.
> 
> Maybe you can help with some simple planning. I usually start with a simple list of the changes. How accurate is this list (I'm basing this on the table of contents that's been shown):
> 
> 1. A bunch of new spaceframes
> 2. New talents (?)
> 3. New weapons (?)
> 4. Spaceframe creation rules
> 5. Optional ship rules (how many? Do they affect ship creation)
> 6. Space station rules
> 
> Is there other stuff?

1. Yes, multiple spaceframes including a few from STO 
2. Various new talents
3. I didn't see any new weapons, just more detail of existing weapons
4. Yes detailed rules
5. there are a few more mission profiles 
6. yes and space station spaceframes
7. New mission pods for NEbula and other classes that support pods
8. Multiple Small Craft Spaceframes 
9. an optional rule on jury rigging 
10. some mission briefs (not that that affects the creator)

**Comment by bcholmes** [comment]
Per Felderburg's comment, above, If anyone finds a good list of which spaceframes have been updated, that'd be a great help.

I've got a coupl'a thoughts about how to represent the new spaceframes, internally. My take is that if you've included the Utopia Planitia source, the you should get the "latest" write-up for the spaceframe. There are some implications for this -- largely around the way we create the bookmarkable versions of the starships -- but I think those implications are manageable.

**Reply by Felderburg** [reply]
Is it possible to just have two named versions, one is (e.g.) Steamrunner, and the other is Steamrunner (UP)? Or Ambassador (Legacy) and Ambassador? I don't know what goes on internally in the tool (I guess I could look but I don't know JavaScript) but it seems like that may be a decent way to do it.

I think using a "legacy" type naming convention would work best, since I assume (perhaps erroneously) that future spaceframes, both official and constructed, will be made with UP's rules.

Edit: I looked, seems like it's just a list of names in spaceframes.ts. So depending on what sort of names typescript allows (with spaces, underscores, parentheses?) I think just having two named versions would work for me. But I think that's the sort of thing you might want more input on, since other people might have different ideas of how to display the difference between legacy and UP frames.

**Reply by bcholmes** [reply]
Just thinking about this idea now, it is entirely possible but I was sort of allergic to the idea because anything that makes the lists longer usually causes people to feel overwhelmed.

Now, having said that, I suppose I could provide a ticky-box that allows people to include the pre-Utopia Planitia options.

That does raise weird edge cases: should I allow someone to use an "old" Luna spaceframe with new Mission Pods, for example?

**Reply by AExusiai** [reply]
Concidering that the Old Luna in the Command Division Includes the Sensor pod that is something that should be addressed. 

Personally, I'd prefer to have just the updated spaceframes (UP Only) 

LIST OF CHANGED FRAMES 
Luna 
Daedalus
NX Class
Constitution
Miranda
Excelsior
Constellation
Akira
Defiant
Intrepid
Oberth
Centaur
Ambassador
Nebula
Olympic
Steamrunner
Norway
Sovereign
Luna

**Reply by AExusiai** [reply]
> Is it possible to just have two named versions, one is (e.g.) Steamrunner, and the other is Steamrunner (UP)? Or Ambassador (Legacy) and Ambassador? I don't know what goes on internally in the tool (I guess I could look but I don't know JavaScript) but it seems like that may be a decent way to do it.
> 
> I think using a "legacy" type naming convention would work best, since I assume (perhaps erroneously) that future spaceframes, both official and constructed, will be made with UP's rules.
> 
> Edit: I looked, seems like it's just a list of names in spaceframes.ts. So depending on what sort of names typescript allows (with spaces, underscores, parentheses?) I think just having two named versions would work for me. But I think that's the sort of thing you might want more input on, since other people might have different ideas of how to display the difference between legacy and UP frames.

Just an opinion, but if you go with both options, then it is going to be unbalanced. I think any changes (and this is just me) to make things simpler, is just to update the spaceframes that have been updated (see the list above) and have no option for "pre UP" It looks as if UP did a lot of work to balance space frames.

**Reply by Felderburg** [reply]
I think the biggest concern for me was people making a ship, comparing it to other people who built based off the "same" spaceframe and getting confused as to why stats don't match. But I guess at least someone at a table would be aware of the change, and a table is probably just going to use 1 version of the stats, so it might not matter.

I think as long as there is a note somewhere about which version is being used, it's fine.

**Reply by AExusiai** [reply]
> I think the biggest concern for me was people making a ship, comparing it to other people who built based off the "same" spaceframe and getting confused as to why stats don't match. But I guess at least someone at a table would be aware of the change, and a table is probably just going to use 1 version of the stats, so it might not matter.
> 
> I think as long as there is a note somewhere about which version is being used, it's fine.

I guess I think everyone is like me, new ruleset comes out I remake stuff using it.

**Reply by Felderburg** [reply]
Well, some people might not know about it, or might not have the money to buy the pdf.

**Reply by Felderburg** [reply]
In case it's relevant, I made an excel spreadsheet of spaceframes several months ago: https://docs.google.com/spreadsheets/d/131s5pROXg3u-z4JLWJVEaeIMF79vXuPH/ Given your "gonna roll my sleeves up" post on the modiphius forums, I thought it might be helpful.

I think the main reason I did it was to make pods for the Luna class, and also because I was generally curious about the things I measured, and general comparisons. Given what I've read about the UP custom frame construction, I am academically curious how the rules there compare to the broader stats.

**Reply by bcholmes** [reply]
Thanks, @AExusiai: the list you provided proved to be very helpful.

**Reply by AExusiai** [reply]
> Thanks, @AExusiai: the list you provided proved to be very helpful.

Glad I could help. 

Observation, The Constitution Sauser Sep I don't think it takes a Talent slot, and the creator seems to have it take up a slot. In the book it's listed as a special rule, not a talent
